tenants-certified-attributes-updater/src/main/scala/it/pagopa/interop/tenantscertifiedattributesupdater/Dependencies.scala

```diff
@@ -0,0 +1,163 @@
+import scala.util.Try
+
+trait Dependencies {
+
+  implicit val contexts: Seq[(String, String)] = Seq.empty
+
+  final val initPage: Int       = 1
+  final val maxLimit: Int       = 100
+  final val groupDimension: Int = 1000
+
+  val jwtConfig: JWTInternalTokenConfig = JWTConfiguration.jwtInternalTokenConfig
+
+  def partyRegistryProxyService(
+    blockingEc: ExecutionContextExecutor
+  )(implicit actorSystem: ActorSystem[_]): PartyRegistryProxyServiceImpl =
+    PartyRegistryProxyServiceImpl(
+      PartyRegistryProxyInvoker(blockingEc)(actorSystem.classicSystem),
+      InstitutionApi(ApplicationConfiguration.partyRegistryProxyURL)
+    )(blockingEc)
+
+  def tenantProcessService(
+    blockingEc: ExecutionContextExecutor
+  )(implicit actorSystem: ActorSystem[_]): TenantProcessServiceImpl =
+    TenantProcessServiceImpl(
+      TenantProcessInvoker(blockingEc)(actorSystem.classicSystem),
+      TenantApi(ApplicationConfiguration.tenantProcessURL)
+    )(blockingEc)
+
+  def signerService(blockingEc: ExecutionContextExecutor): SignerService = new KMSSignerService(blockingEc)
+
+  def generateBearer(
+    blockingEc: ExecutionContextExecutor
+  )(implicit actorSystem: ActorSystem[_], ec: ExecutionContext): Future[String] = for {
+    tokenGenerator <- interopTokenGenerator(blockingEc)
+    m2mToken       <- tokenGenerator
+      .generateInternalToken(
+        subject = jwtConfig.subject,
+        audience = jwtConfig.audience.toList,
+        tokenIssuer = jwtConfig.issuer,
+        secondsDuration = jwtConfig.durationInSeconds
+      )
+  } yield m2mToken.serialized
+
+  def interopTokenGenerator(
+    blockingEc: ExecutionContextExecutor
+  )(implicit actorSystem: ActorSystem[_], ec: ExecutionContext): Future[InteropTokenGenerator] =
+    Try(
+      new DefaultInteropTokenGenerator(
+        signerService(blockingEc),
+        new PrivateKeysKidHolder {
+          override val RSAPrivateKeyset: Set[KID] = ApplicationConfiguration.rsaKeysIdentifiers
+          override val ECPrivateKeyset: Set[KID]  = ApplicationConfiguration.ecKeysIdentifiers
+        }
+      )
+    ).toFuture
+
+  def createAction(
+    institutions: List[Institution],
+    tenants: List[PersistentTenant],
+    attributesIndex: Map[UUID, AttributeInfo]
+  ): TenantActions = {
+    val fromRegistry: Map[PersistentExternalId, AttributeInfo] =
+      institutions
+        .filter(_.id.nonEmpty)
+        .map { institution =>
+          PersistentExternalId(institution.origin, institution.id) -> AttributeInfo(
+            institution.origin,
+            institution.category
+          )
+        }
+        .toMap
+
+    val fromTenant: Map[PersistentExternalId, List[AttributeInfo]] =
+      tenants.map(tenant => tenant.externalId -> tenant.attributes.flatMap(attr => attributesIndex.get(attr.id))).toMap
+
+    val activations: List[InternalTenantSeed] =
+      fromRegistry
+        .filter(i => !fromTenant.contains(i._1))
```

Comment:
`filter(_ => !p)` is pretty much unreadable, prefer using (here and a few lines down) `filterNot(_ => p)` that is standard library api too